### PR TITLE
Elasticsearch index creation

### PR DESCRIPTION
**Pull request by kevinstyp**

### CODE REVIEW

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 1]
```diff
@@ -113,6 +122,72 @@ private IndexRequest generateIndexRequest(Sample sample) throws IOException {
...
+        logger.log(Level.INFO, String.format("indexExists: %s", indexExists));
+        if (!indexExists) {
+            CreateIndexRequest requestCreate = new CreateIndexRequest(indexName);
+            //requestCreate.
+
+            XContentBuilder builder = XContentFactory.jsonBuilder();
+            builder.startObject();
+            // Brackets only for readability
+            {
+                builder.startObject("_doc");
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: please declare as static variable

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 2]
```diff
@@ -113,6 +122,72 @@ private IndexRequest generateIndexRequest(Sample sample) throws IOException {
...
+                {
+                    builder.startObject("properties");
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: please declare as static variable

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 3]
```diff
@@ -113,6 +122,72 @@ private IndexRequest generateIndexRequest(Sample sample) throws IOException {
...
+                    {
+                        // Create timestamp field as date which is saved in milliseconds
+                        builder.startObject("timestamp");
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: please declare as static variable

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 4]
```diff
@@ -113,6 +122,72 @@ private IndexRequest generateIndexRequest(Sample sample) throws IOException {
...
+                        {
+                            builder.field("type", "date");
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: please declare as static variable

**Review by flohannes — CHANGES_REQUESTED** [review]

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 5]
```diff
@@ -113,6 +122,72 @@ private IndexRequest generateIndexRequest(Sample sample) throws IOException {
...
+                logger.log(Level.INFO, String.format("Created index for indexName '%s' and got response '%s' and '%s'.", indexName, acknowledged, shardsAcknowledged));
+            }
+            catch (ElasticsearchStatusException elasticEx){
+                //Can happen if multiple forks try to create this index, just ignore the message that it already was added
+                if (!elasticEx.getMessage().contains("resource_already_exists_exception")){
+                    throw elasticEx;
+                }
+            }
+        }
+        indexCheckDone = true;
```
Comment (antongulenko) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: Please move this assignment into the if clause in generateIndexRequest(), to have the check and assignment together.

**Review by flohannes — APPROVED** [review]

**Pull request merged**